
Fix reported PHY temperature by masking the upper word.

Report & discussion:
https://lore.kernel.org/netdev/0db14339-1b69-8fa4-21fd-6d436037c945@applied-asynchrony.com/

--- a/drivers/net/ethernet/aquantia/atlantic/hw_atl/hw_atl_utils_fw2x.c
+++ b/drivers/net/ethernet/aquantia/atlantic/hw_atl/hw_atl_utils_fw2x.c
@@ -337,7 +337,7 @@ static int aq_fw2x_get_phy_temp(struct aq_hw_s *self, int *temp)
 	/* Convert PHY temperature from 1/256 degree Celsius
 	 * to 1/1000 degree Celsius.
 	 */
-	*temp = temp_res  * 1000 / 256;
+	*temp = (temp_res & 0xFFFF)  * 1000 / 256;
 
 	return 0;
 }
